modules/cog-fdo-shell.c

```diff
@@ -30,9 +30,12 @@ typedef struct {
     CogShell parent;
 } CogFdoShell;
 
+static void cog_fdo_shell_initable_iface_init (GInitableIface *iface);
+
 G_DEFINE_DYNAMIC_TYPE_EXTENDED (CogFdoShell, cog_fdo_shell, COG_TYPE_SHELL, 0,
     g_io_extension_point_implement (COG_MODULES_SHELL_EXTENSION_POINT,
-                                    g_define_type_id, "fdo", 100))
+                                    g_define_type_id, "fdo", 100);
+    G_IMPLEMENT_INTERFACE_DYNAMIC (G_TYPE_INITABLE, cog_fdo_shell_initable_iface_init))
```

Comment:
For ref:  https://developer.gnome.org/gobject/stable/GTypeModule.html#G-DEFINE-DYNAMIC-TYPE-EXTENDED:CAPS

This is the `CODE` part of the `G_DEFINE_DYNAMIC_TYPE_EXTENDED`, the custom code that gets inserted in the *_get_type() function. This leads in a  `g_type_module_add_interface ()`  which logic will be implemented by the  `cog_fdo_shell_initable_iface_init()` during the registration of the type.